### PR TITLE
Migrate to the new parent POM

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Migrate to the new parent POM
- [x] Bump the code dependency to 1.532.3 
- [x] Jelly escaping
- [x] Fix FindBugs and Javadoc issues

CC @KostyaSha

### CODE REVIEW

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java [hunk 1]
```diff
@@ -128,12 +128,49 @@ public final Boolean hasAnyPermission(Set<Permission> permissions) {
...
+   * Used to sort the sets. 
+   * We presume that any role name is being used once and only once.
    * @param o The object you want to compare this instance to
    * @return Comparison of role names
    */
+  @Override
   public int compareTo(Object o) {
-    return this.name.compareTo(((Role) o).getName());
+    if (o instanceof Role) {
+        return name.compareTo(((Role)o).name);
```
Comment (KostyaSha) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: space before o :)

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java [hunk 2]
```diff
@@ -128,12 +128,49 @@ public final Boolean hasAnyPermission(Set<Permission> permissions) {
...
+    }
+    return -1;
   }
 
+    @Override
+    public int hashCode() {
```
Comment (KostyaSha) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: I would suggest just pick HashCodeBuilder

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java [hunk 3]
```diff
@@ -128,12 +128,49 @@ public final Boolean hasAnyPermission(Set<Permission> permissions) {
...
+        int hash = 7;
+        hash = 53 * hash + (this.name != null ? this.name.hashCode() : 0);
+        hash = 53 * hash + (this.pattern != null ? this.pattern.hashCode() : 0);
+        hash = 53 * hash + (this.permissions != null ? this.permissions.hashCode() : 0);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (KostyaSha) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: And EqualsBuilder :)

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java [hunk 4]
```diff
@@ -128,12 +128,49 @@ public final Boolean hasAnyPermission(Set<Permission> permissions) {
...
+            return false;
+        }
+        if (this.pattern != other.pattern && (this.pattern == null || !this.pattern.equals(other.pattern))) {
+            return false;
+        }
+        if (this.permissions != other.permissions && (this.permissions == null || !this.permissions.equals(other.permissions))) {
+            return false;
+        }
+        return true;
+    }
```
Comment (KostyaSha) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: And ToStringBuilder :)

File: src/test/java/com/synopsys/arc/jenkins/plugins/rolestrategy/MacroTest.java [hunk 5]
```diff
@@ -131,7 +131,7 @@ private static Macro assertParseMacroDriver(String errorMessage,
         Macro res = null;
         try 
         {
-            res = Macro.Parse(macroString);
+            res = Macro.parse(macroString);
```
Comment (KostyaSha) [inline src/test/java/com/synopsys/arc/jenkins/plugins/rolestrategy/MacroTest.java]: :)

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: sure. I just use code gen in my IDE, but oh well

Comment (KostyaSha) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: Problem that i always missing all this updates when adding new field. So mostly switched to reflection and periodically finding real bugs detected by tests that using tostring/equals/hashcode.

Comment (oleg-nenashev) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: I decided not to use reflection-based methods. Performance may become a serious problem in this case

**Comment by oleg-nenashev** [comment]
CC @reviewbybees

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java (hunk 1 above)
Comment (recena) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/Role.java]: 👍

**Comment by recena** [comment]
🐝 and 👍

**Pull request merged**